torch/csrc/jit/passes/onnx/list_model_parameters.cpp

```diff
@@ -0,0 +1,164 @@
+#include <torch/csrc/jit/passes/onnx/list_model_parameters.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/dead_code_elimination.h>
+
+namespace torch {
+namespace jit {
+
+// findSubModuleAttr function chases getAttr chains to locate the submodules.
+// For example:
+// module M {
+//   attributes {
+//     A = <SubModule at ...>
+//   }
+//   ...
+//   %A = prim::GetAttr[name="A"](%self)
+//   ...
+//   %B = prim::GetAttr[name="B"](%A)
+//   ...
+//   %weight = prim::GetAttr[name="scale"](%B)
+//   ...
+std::deque<std::string> findSubModuleAttr(
+    Value* input,
+    std::string& name,
+    Module& attrModule,
+    std::shared_ptr<Graph>& graph) {
+  Node* node = input->node();
+  std::deque<std::string> moduleNames;
+
+  while (!(node->outputs()[0]->type() == graph->inputs()[0]->type())) {
```

Comment:
```suggestion
  while (node->outputs().at(0)->type() != graph->inputs().at(0)->type()) {
```
`at` is safer as it checks out of bounds

Reply:
Could you explain what this condition is guarding against?